worlds/sm/__init__.py

```diff
@@ -701,8 +700,8 @@ def write_spoiler(self, spoiler_handle: TextIO):
                                                             dest.Name) for src, dest in self.variaRando.randoExec.areaGraph.InterAreaTransitions if src.Boss]))
 
 def create_locations(self, player: int):
-    for name, id in locations_lookup_name_to_id.items():
-        self.locations[name] = SMLocation(player, name, id)
+    for name in locationsDict.keys():
```

Comment:
```suggestion
    for name in locationsDict:
```
.keys() is not needed here.